test/yarp/comments_test.rb

```diff
@@ -74,19 +75,40 @@ def bar
       method_node = class_node.body.body.first
       call_node = method_node.body.body.first
 
-      assert_equal("# Foo class\n# Foo end\n", class_node.location.comments.map { |c| c.location.slice }.join)
-      assert_equal("# bar method\n# bar end\n", method_node.location.comments.map { |c| c.location.slice }.join)
-      assert_equal("# baz invocation\n", call_node.location.comments.map { |c| c.location.slice }.join)
+      assert_equal("# Foo class\n# Foo end", class_node.location.comments.map { |c| c.location.slice }.join("\n"))
+      assert_equal("# bar method\n# bar end", method_node.location.comments.map { |c| c.location.slice }.join("\n"))
+      assert_equal("# baz invocation", call_node.location.comments.map { |c| c.location.slice }.join("\n"))
     end
 
     private
 
-    def assert_comment(source, type, location)
+    def assert_comment(source, type, locations)
+      start_offset, end_offset, start_line, end_line, start_column, end_column = locations
+      expected = {
```

Comment:
We changed to use a hash so that the failure would report the full set of differences in location, rather than stopping at the first difference.